trpcultivate_genotypes/tests/src/Kernel/GenotypesLoader/GenotypesLoaderProcessSamplesTest.php

```diff
@@ -0,0 +1,401 @@
+	}
+
+  /**
+   * Test processing a samples file with 7 columns, to simulate a real-life example
+   *
+   * @group GenotypesLoader
+   */
+  public function testProcessSamplesSevenColumns(){
+
+		// Assert the plugin was created
+		$this->assertNotNull($this->plugin);
+
+		// Sample Filepath
+		$sample_file_path = __DIR__ . '/../../Fixtures/cats_samples.tsv';
+
+		// Set sample filepath
+		$success = $this->plugin->setSampleFilepath($sample_file_path);
+		$this->assertTrue($success, "Unable to set sample filepath");
+
+		// Get sample filepath
+		$grabbed_sample_file_path = $this->plugin->getSampleFilepath();
+		$this->assertEquals($sample_file_path, $grabbed_sample_file_path, "The sample filepath grabbed by the getter method does not match.");
+
+		// Insert our 2 organisms that are in the file
+		// Felis catus
+		$catus_organism_id = $this->connection->insert('1:organism')
+			->fields([
+				'genus' => 'Felis',
+				'species' => 'catus',
+			])
+			->execute();
+
+		// Felis silvestris
+		$silvestris_organism_id = $this->connection->insert('1:organism')
+			->fields([
+				'genus' => 'Felis',
+				'species' => 'silvestris',
+			])
+			->execute();
+
+		// Process our samples so that they all get inserted into the database
+		$processed_samples = $this->plugin->processSamples();
+
+		// Setup our array with our samples and compare it to the output from our method
+		$samples_array = [
+			'Ross' => 1,
+			'Prado' => 3,
+			'Ash' => 5,
+			'Piero' => 7,
+			'Tai' => 9,
+			'Beverly' => 11,
+			'Argent' => 13,
+			'Trenus' => 15,
+			'Zapelli' => 17
+		];
+
+		// Check that the number of stocks match what we expect
+    $this->assertEquals(count($samples_array), count($processed_samples), "The number of samples that were processed is incorrect.");
+
+		// Compare our returned samples array with what we expect to get
+		$this->assertEquals($samples_array, $processed_samples, "The returned samples array is not what was expected.");
+
+		// Check that our samples are the correct organisms
+		// First check Ross is a Felis catus
+		$Ross_query = $this->connection->select('1:stock','s')
+    	->fields('s', ['organism_id'])
+		  ->condition('stock_id', 1, '=');
+		$Ross_record = $Ross_query->execute()->fetchAll();
+		$this->assertEquals($catus_organism_id, $Ross_record[0]->organism_id, "One of the samples that was inserted (Ross) is of the wrong organism.");
+
+		// Second, check Zapelli is a Felis silvestris
+		$Zapelli_query = $this->connection->select('1:stock','s')
+    	->fields('s', ['organism_id'])
+		  ->condition('stock_id', 17, '=');
+		$Zapelli_record = $Zapelli_query->execute()->fetchAll();
+		$this->assertEquals($silvestris_organism_id, $Zapelli_record[0]->organism_id, "One of the samples that was inserted (Zapelli) is of the wrong organism.");
+
+		// Check that our samples' germplasm are the correct germplasm type
+		// Pull out the cvterm ID for CO_010:0000044
```

Comment:
There's actually a testing API function for pulling out a cvterm_id based on the accession: https://github.com/tripal/tripal/blob/4.x/tripal_chado/tests/src/Functional/ChadoTestTrait.php#L68-L86. This function can be used in both kernel and functional tests.

Reply:
Implemented in commit [9aea15b](https://github.com/TripalCultivate/TripalCultivate-Genetics/pull/26/commits/9aea15b06542f9e8bd23e223657733fe352bd377) and also added when I added more tests in commit [5e17f31](https://github.com/TripalCultivate/TripalCultivate-Genetics/pull/26/commits/5e17f31a9adeefe331600bb46fd6018d871b31d1)